### ISSUE TITLE
Accessibility: Content is not enclosed within a landmark region.

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52145]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

This visual section of the page is not programmatically defined as a landmark region. When content is located outside of a landmark region it can be difficult for screen reader users to navigate to and find.

![Image](https://github.com/user-attachments/assets/20c8d0be-11bb-43bb-ba8a-df897ff2ca98)

### Component or Page: 

WooCommerce Store Notice

### Name affected element:

Container for Store Notice

### Relevant code snippets:

`<p class="woocommerce-store-notice demo_store" data-notice-id="fe68900dee1653cf2d0662bb68d31594" style="display: block;">This is a demo store for testing purposes — no orders shall be fulfilled. <a href="#" class="woocommerce-store-notice__dismiss-link">Dismiss</a></p>`

### Recommended fix:

Include this section in a landmark region. This could either be located in the header as noted in #WooCore-078 or be in a section tag with an accessible name using aria-labelledby or aria-label. The section tag is likely easier to implement. The aria-label can just be "Store Notice".

See https://www.w3.org/WAI/ARIA/apg/practices/landmark-regions/

**Comment by Manussakis** [comment]
The store notice is a paragraph element that [has been given the `role="dialog"`](https://github.com/woocommerce/woocommerce/pull/55494/files#diff-d49488c34017d273db858b7b6eff58f22d95310991d26638e75793c6a1e98678R1099) attribute to fix the #52144 issue.
This means that updating its role to turn it into a live region seems no longer an option.

The next approach could be wrapping the store notice in a `<section>` tag, however, the notice element if filtered by the [`woocommerce_demo_store` filter](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-template-functions.php#L1093) and updating its markup is a source of regressions.

After thinking about this issue, I realized that [moving the store notice to the top of the page](https://github.com/woocommerce/woocommerce/pull/55494/files#diff-2ad0aafe6a35b7a71dd6544239ea169b909c797677fb8edd52c011fef41ab977R258) was also a fix for the #52144 issue.
This means that it will be the first element on the page voiced by screen readers once focused. (even before the "Skip to navigation" link).

One of the reasons to organize the page content with landmarks regions is to allow screen reader users to identify the structure of a page and easily navigate through it.
Ideally, all the content should be inside a landmark region, but considering that:
- The store notice will be voiced first.
- It can be dismissed.
- Users are unlikely to navigate back to it once its message is announced.

Leaving it outside a landmark region to avoid dealing with regressions seems acceptable to me.

If we do want to wrap the notice in a `<section>` tag, I suggest creating a setting on the admin so users can control the markup update. This final option seems an overkill to me, TBH.

Amber Hinds (I was not able to tag you), can you think of another alternative to fix it 🤔

I also would like to hear the Woo team's thoughts.

cc: @vikrampm1

**Comment by vikrampm1** [comment]
cc @amberhinds for your thoughts on the above.

**Comment by amberhinds** [comment]
Changing the notice to a dialog is a sufficient fix for this, so long as the dialog is modal. 

From the ARIA Authoring Practice Guidelines:
>Note that wrapping the content of a modal dialog in a landmark region is unnecessary. A landmark that wraps modal content cannot provide any benefit to users because it is not perceivable unless the modal is open. In addition, when a modal is open, a landmark containing its content is superfluous because the modal itself is a container that provides both a name and boundaries.

It looks like in [this commit ](https://github.com/woocommerce/woocommerce/pull/55494/commits/b3e7be289e23e649d70bb3c18e1012910b01f4ac)that only the dialog role was added (not aria-modal). I want to listen to it to confirm this is sufficient to also resolve this issue.

Was there any internal discussion about whether the store notice should be treated as a modal? I don't see any on https://github.com/woocommerce/woocommerce/issues/52144

**Comment by Manussakis** [comment]
Hey @amberhinds 
There wasn't any discussion.

Changing the Store Notice behavior to make it act like a modal is a huge effort, requires design inputs, and is prone to regressions, (for the same reason flagged in [this comment](https://github.com/woocommerce/woocommerce/issues/52145#issuecomment-2659813802), unless we build it from scratch and don't update the current markup). I see it as a breaking change.

Since the other option seemed to fix the issue with less impact on the code base, I went for it.

Could you please confirm if that is sufficient? I am happy to change the approach but would be great to hear from the Woo 
@woocommerce/proton  team first.

Thanks!

cc: @vikrampm1

**Comment by amberhinds** [comment]
When I say it needs to be modal, I mean it needs aria-modal="true" on it, and it would have to be dismissed before the rest of the page could be interacted with. The design wouldn't need to be changed at all.

**Comment by Manussakis** [comment]
> When I say it needs to be modal, I mean it needs aria-modal="true" on it, and it would have to be dismissed before the rest of the page could be interacted with. The design wouldn't need to be changed at all.

I see, @amberhinds, thanks for the clarification.
So in this case, if we don't touch the design at all, such as adding an overlay where the banner would be on, we would be causing an issue for non-screen-reader users. Without any design changes, they would try to interact with the content as they normally do, but with no success, until they realized they needed to dismiss the banner first.

>  I want to listen to it to confirm this is sufficient to also resolve this issue.

Did you have a chance to test it?

**Comment by vikrampm1** [comment]
@amberhinds just following up on the above if you have any further thoughts before we proceed with the implementation.

**Comment by amberhinds** [comment]
Hi @vikrampm1, thanks for your patience. I'll have feedback for you tomorrow. I need to get this change made on our copy for testing.

**Comment by amberhinds** [comment]
This treatment is not revealing the content in the landmarks list in VoiceOver.

![Image](https://github.com/user-attachments/assets/ed3b6125-15f9-43c4-a0c3-82378dc8a138)

We discussed this internally and we think that if we're not treating it like a modal it should have a complementry role instead and an aria label, like this:
`<p class="woocommerce-store-notice demo_store" data-notice-id="fe68900dee1653cf2d0662bb68d31594" style="display: block;" role="complementary" aria-label="store notice"> [...] </p>`

**Comment by Manussakis** [comment]
@amberhinds 
Great solution! The only change I made is to add the notice message as the `aria-label` value instead of "Store notice", otherwise, the message wouldn't be voiced since the `aria-label` attribute gets precedence over the element text.

**Comment by amberhinds** [comment]
@Manussakis the aria-label on a non-interactive component like a landmark region is not going to override the content within it. For example, an aria-label of primary on a nav tag does not stop the user from hearing the content of the navigation menu within it.

Here's a video showing what it sounds like with the aria-label being set to "store notice". 

https://github.com/user-attachments/assets/a1d6c53e-ffb9-42db-a8b9-767ae2b066ff

It's best to have a concise label for a landmark so it's easier to understand when navigating by landmarks. Can you please set it to "store notice"?

**Comment by Manussakis** [comment]
Thanks for clarifying that, @amberhinds .
Updated!
